Zeeguu Reader/Zeeguu Reader/ArticleTableViewCell.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  ArticleTableViewCell.swift
+//  Zeeguu Reader
+//
+//  Created by Jorrit Oosterhof on 27-01-16.
+//  Copyright © 2015 Jorrit Oosterhof.
+// 
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//  
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//  
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import UIKit
+import ZeeguuAPI
+
+class ArticleTableViewCell: UITableViewCell {
+	
+	private var article: Article
+	
+	private var titleField: UILabel
+	private var descriptionField: UILabel
+	private var articleImageView: UIImageView
+	
+	init(article: Article, reuseIdentifier: String?) {
+		self.article = article
+		titleField = UILabel.autoLayoutCapapble()
+		descriptionField = UILabel.autoLayoutCapapble()
+		articleImageView = UIImageView.autoLayoutCapapble()
+		super.init(style: .Default, reuseIdentifier: reuseIdentifier)
+		setupLayout()
+		updateLabels()
+	}
+	
+	required init?(coder aDecoder: NSCoder) {
+		fatalError("init(coder:) has not been implemented")
```

Comment:
See the link. Basically, it indicates that the initializer might fail. That means that the initializer may return `nil` to indicate failure and therefore, the return value is an `Optional`.

This initializer is part of `UIKit` and is required.

(In other classes I avoided this by making all properties `Optional` and creating a convenience initializer that gave the properties their value. However, if you want non-`Optional` properties, then a designated initializer is needed (which is `init(article:reuseIdentifier)`, but the `init?(coder:)` is also needed (otherwise, Xcode complains and won't compile). That's how the language is built.)

http://www.apeth.com/swiftBook/ch04.html#_failable_initializers